applications/IGAStructuralMechanicsApplication/custom_elements/base_discrete_element.cpp

```diff
@@ -0,0 +1,559 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/IGAStructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//
+
+
+// System includes
+#include "includes/define.h"
+#include "utilities/math_utils.h"
+
+// External includes
+
+// Project includes
+#include "custom_elements/base_discrete_element.h"
+
+#include "iga_structural_mechanics_application.h"
+#include "iga_structural_mechanics_application_variables.h"
+
+#include "geometries/geometry.h"
+
+namespace Kratos
+{
+    void BaseDiscreteElement::Initialize()
+    {
+        KRATOS_TRY
+
+        //Constitutive Law initialisation
+        if (mConstitutiveLawVector.size() != 1)
+            mConstitutiveLawVector.resize(1);
+
+        InitializeMaterial();
+
+        KRATOS_CATCH("")
+    }
+
+    /***********************************************************************************/
+    /***********************************************************************************/
+    void BaseDiscreteElement::InitializeMaterial()
+    {
+        KRATOS_TRY
+
+            if (GetProperties()[CONSTITUTIVE_LAW] != nullptr) {
+                mConstitutiveLawVector[0] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
+                mConstitutiveLawVector[0]->InitializeMaterial(GetProperties(),
+                    GetGeometry(),
+                    GetValue(SHAPE_FUNCTION_VALUES)
+                );
+            }
+            else
```

Comment:
brackets

Reply:
I would say it makes it much more readable

Reply:
Just an advice... I don't think that the readability changes but leaving them off can lead to semantic errors in the code.